sqlmesh/dbt/loader.py

```diff
@@ -150,7 +150,10 @@ def _load_project(self) -> Project:
             )
 
         for package_name, macro_infos in context.manifest.all_macros.items():
-            context.jinja_macros.add_macros(macro_infos, package=package_name)
+            context.add_macros(macro_infos, package=package_name)
+
+        # TODO this needs to make its way into engine adapter's default database
+        context.default_database = context.render("{{ generate_database_name() }}")
```

Comment:
I'd rather call `jinja_macros` directly and not bother with `Context.add_macros`
```
context.jinja_macros.build_macro(MacroReference(name="generate_database_name"))()
```
Feel free to have a helper for this in the context. Or better yet just make it a proeprty method of the `Context` instead:
```
@property
def default_database(self):
    return self.jinja_macros.build_macro(MacroReference(name="generate_database_name"))()
```

Reply:
This way you can also do a proper error handling if the `generate_database_name` macro is missing.

Reply:
Good idea!